### ISSUE TITLE
Enclose post author fields in Author object in Post Response

### ISSUE THREAD

**Issue by rachelbaker** [WP-API/WP-API#867]
A regression we have in develop is to only expose the author_id, which is pretty useless when it comes to displaying a Post and crediting the post author.  

My suggestion here is to create an author object, that encloses the following fields:

  - id (author id)
  - name (author display name)
  - url (author url)
  - avatar (author avatar url)

Example:

```json
  "author": {
        "id": 1,
        "name": "Rachel",
        "url": "http:\/\/10up.com",
        "avatar": "https:\/\/1.gravatar.com\/avatar\/767fc9c115a1b989744c755db47feb60?s=96&d=retro",
    },
```

**Comment by rachelbaker** [comment]
Based on a discussion with @rmccue the new idea is to turn on embedding by default, and embed the author in `_links`.  The author object would be accessible from the `_embedded.author` object.

**Comment by danielbachhuber** [comment]
How do we decide which connected objects to embed by default? For the convenience argument, featured image seems like another likely candidate.

**Comment by rmccue** [comment]
> How do we decide which connected objects to embed by default? For the convenience argument, featured image seems like another likely candidate.

We have the `embedded` link attribute that indicates whether they *can* be embedded; likely all of these can be pulled in without an issue.

**Comment by danielbachhuber** [comment]
This has been addressed with #1022
